src/service/main.cpp

```diff
@@ -153,27 +155,51 @@ int main(int argc, char **argv) {
     using GrFgWorker  = GnuRadioFlowGraphWorker<GrAcqWorker, "/flowgraph", description<"Provides access to the GnuRadio flow graph">>;
     gr::BlockRegistry registry;
     registerTestBlocks(registry);
-    gr::PluginLoader pluginLoader(registry, {});
-    GrAcqWorker      grAcqWorker(broker, &pluginLoader, std::chrono::milliseconds(50));
-    GrFgWorker       grFgWorker(broker, &pluginLoader, { grc, {} }, grAcqWorker);
-
-    std::jthread     grAcqWorkerThread([&grAcqWorker] { grAcqWorker.run(); });
-    std::jthread     grFgWorkerThread([&grFgWorker] { grFgWorker.run(); });
-
-    std::this_thread::sleep_for(100ms);
+    gr::PluginLoader                                          pluginLoader(registry, {});
+    GrAcqWorker                                               grAcqWorker(broker, &pluginLoader, std::chrono::milliseconds(50));
+    GrFgWorker                                                grFgWorker(broker, &pluginLoader, { grc, {} }, grAcqWorker);
 
     const opencmw::zmq::Context                               zctx{};
     std::vector<std::unique_ptr<opencmw::client::ClientBase>> clients;
     clients.emplace_back(std::make_unique<opencmw::client::MDClientCtx>(zctx, 20ms, ""));
     clients.emplace_back(std::make_unique<opencmw::client::RestClient>(opencmw::client::DefaultContentTypeHeader(opencmw::MIME::BINARY)));
     opencmw::client::ClientContext client{ std::move(clients) };
 
-    // create example signals
-    opencmw::service::dns::DnsClient dns_client{ client, settings.serviceUrls() + "/dns" };
-    dns_client.registerSignals({ { "http", "localhost", 8080, "opendigitizer", "", "Signal A", "A", 1e3, "TRIGGERED" },
-            { "http", "localhost", 8080, "opendigitizer", "", "Signal B", "A", 2e3, "STREAMING" },
-            { "https", "powersupply.example.com", 8080, "powersupply", "", "Supply", "V", 3e3, "TRIGGERED" },
-            { "mdp", "dcct.example.com", 9999, "opendigitizer", "", "signal 4", "ms", 4e3, "STREAMING" } });
+    dns::DnsClient                 dns_client{ client, "http://localhost:8080/dns" };
+
+    std::vector<SignalEntry>       registeredSignals;
+    grAcqWorker.setUpdateSignalEntriesCallback([&registeredSignals, &dns_client](std::vector<SignalEntry> signals) {
+        std::ranges::sort(signals);
+        std::vector<SignalEntry> toUnregister;
+        std::ranges::set_difference(registeredSignals, signals, std::back_inserter(toUnregister));
+        std::vector<SignalEntry> toRegister;
+        std::ranges::set_difference(signals, registeredSignals, std::back_inserter(toRegister));
+
+        auto dnsEntriesForSignal = [](const SignalEntry &entry) {
+            // TODO do smarter host name and port, publish acquisition modes other than streaming?
+            return std::vector{
+                dns::Entry{ "http", "localhost", 8080, "/GnuRadio/Acquisition", "", entry.name, entry.unit, entry.sample_rate, "STREAMING" },
```

Comment:
This will need fixing, at least the hostname must not be hardcoded.

Reply:
Agree ... since you are off, this (and other locations) should be done in a follow-up PRs.